data-prepper-plugins/otel-trace-source/src/test/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceSourceTest.java

```diff
@@ -159,4 +190,79 @@ public void testStartWithEmptyBuffer() {
         final OTelTraceSource source = new OTelTraceSource(new PluginSetting(null, null));
         Assertions.assertThrows(IllegalStateException.class, () -> source.start(null));
     }
+
+    @Test
+    public void testStartWithServerExecutionExceptionNoCause() throws ExecutionException, InterruptedException {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
+            armeriaServerMock.when(() -> Server.builder()).thenReturn(serverBuilder);
+            when(completableFuture.get()).thenThrow(new ExecutionException("", null));
+            final OTelTraceSource source = new OTelTraceSource(new PluginSetting(null, null));
+            assertThrows(RuntimeException.class, () -> source.start(BUFFER));
+        }
+    }
+
+    @Test
+    public void testStartWithServerExecutionExceptionWithCause() throws ExecutionException, InterruptedException {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
+            armeriaServerMock.when(() -> Server.builder()).thenReturn(serverBuilder);
+            final NullPointerException expCause = new NullPointerException();
+            when(completableFuture.get()).thenThrow(new ExecutionException("", expCause));
+            final OTelTraceSource source = new OTelTraceSource(new PluginSetting(null, null));
+            final RuntimeException ex = assertThrows(RuntimeException.class, () -> source.start(BUFFER));
+            assertEquals(expCause, ex);
+        }
+    }
```

Comment:
nit - personal preference would be to break up this block (and blocks like it) for readability:

```
    @Test
    public void testStartWithServerExecutionExceptionWithCause() throws ExecutionException, InterruptedException {
        // Arrange
        final OTelTraceSource source = new OTelTraceSource(new PluginSetting(null, null));

        final NullPointerException expCause = new NullPointerException();
        when(completableFuture.get()).thenThrow(new ExecutionException("", expCause));

        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
            armeriaServerMock.when(() -> Server.builder()).thenReturn(serverBuilder);

            // Act
            final RuntimeException ex = assertThrows(RuntimeException.class, () -> source.start(BUFFER));
            
            // Assert
            assertEquals(expCause, ex);
        }
    }
```